main.go

```diff
@@ -148,147 +28,30 @@ func failf(format string, v ...interface{}) {
-	// ---
+	var config Configs
 
-	rootBuildGradleParentDir := filepath.Dir(rootBuildGradlePth)
-	gradlewPth := filepath.Join(rootBuildGradleParentDir, "gradlew")
-	if exist, err := pathutil.IsPathExists(gradlewPth); err != nil {
-		failf("Failed to check if gradlew exist at: %s, error: %s", gradlewPth, err)
-	} else if exist {
-		log.Donef("Gradle Wrapper exist at: %s", gradlewPth)
-		return
+	if err := stepconf.Parse(&config); err != nil {
+		failf("Couldn't create step config: %v", err)
 	}
 
-	//
-	// Ensure licenses
+	stepconf.Print(config)
 	fmt.Println()
-	log.Infof("Ensure Android SDK Licenses")
 
-	sdk, err := sdk.New(configs.AndroidHome)
-	if err != nil {
-		failf("Failed to create sdk, error: %s", err)
-	}
-
-	licencesDir := filepath.Join(sdk.GetAndroidHome(), "licenses")
-	if exist, err := pathutil.IsDirExists(licencesDir); err != nil {
-		failf("Failed to check if licences dir exist at: %s, error: %s", licencesDir, err)
-	} else if !exist {
-		log.Printf("licenses dir not exist, generating...")
-		if err := os.Mkdir(licencesDir, 0777); err != nil {
-			failf("Failed to create dir at: %s, error: %s", licencesDir, err)
-		}
-	}
-
-	for licenceKey, licenseValue := range licenseMap {
-		licencePth := filepath.Join(licencesDir, licenceKey)
-		if exist, err := pathutil.IsPathExists(licencePth); err != nil {
-			failf("Failed to check if license exist at: %s, error: %s", licencePth, err)
-		} else if !exist {
-			log.Printf("%s not exist, generating...", licenceKey)
-			if err := fileutil.WriteStringToFile(licencePth, licenseValue); err != nil {
-				failf("Failed to write license, error: %s", err)
-			}
-			log.Donef("%s generated", licenceKey)
-		} else {
-			log.Donef("%s exist", licenceKey)
-		}
-	}
-	// ---
-
-	//
-	// Generate Gradle Wrapper
-	fmt.Println()
-	log.Infof("Generate Gradle Wrapper")
-
-	wrapperDir := filepath.Join(sdk.GetAndroidHome(), "tools", "templates", "gradle", "wrapper")
-	if exist, err := pathutil.IsDirExists(wrapperDir); err != nil {
-		failf("Failed to check if path: %s exists, error: %s", wrapperDir, err)
-	} else if !exist {
-		failf("gradle wrapper template not exists at: %s", wrapperDir)
-	}
-
-	gradlewInSDKPth := filepath.Join(wrapperDir, "gradlew")
-	if err := command.CopyFile(gradlewInSDKPth, gradlewPth); err != nil {
-		failf("Failed to copy gradlew from: %s to: %s", gradlewInSDKPth, gradlewPth)
-	}
-
-	gradleDir := filepath.Join(rootBuildGradleParentDir, "gradle")
-	if err := os.Mkdir(gradleDir, 0777); err != nil {
-		failf("Failed to create: %s, error: %s", gradleDir, err)
-	}
-
-	gradleWrapperDirInSDK := filepath.Join(wrapperDir, "gradle", "wrapper")
-	if err := command.CopyDir(gradleWrapperDirInSDK, gradleDir, false); err != nil {
-		failf("Failed to copy: %s to %s, error: %s", gradleWrapperDirInSDK, gradleDir, err)
+	gradleCommand := command.New("gradle", "wrapper", "--gradle-version", config.GradleVersion)
+	gradleCommand.SetDir(config.ProjectRootDir)
+	if err := gradleCommand.Run(); err != nil {
```

Comment:
Using RunAndReturnCombinedOutput() would help us a lot when there is an error. Sadly, the err here returns only messages like `exited` or `exit status 1` however the tool might wrote some useful to the stdout or stderr channels. I would add that process output also to the error message. Wdyt?

Reply:
sure, thanks for the hint